scripts/pre_commit_linter.py

```diff
@@ -116,22 +117,67 @@ def _get_changed_filenames():
     return unstaged_files + staged_files
 
 
-def _get_all_files_in_directory(dir_path):
+def _filename_match(filename, sources):
+    """Checks if filname matches with source
```

Comment:
Actually, what about removing this altogether and using:

   https://docs.python.org/2/library/fnmatch.html